File: packages/php-sdk/tests/AuthTest.php
<?php

namespace Wusul\Tests;

use PHPUnit\Framework\TestCase;
use Wusul\Auth;

class AuthTest extends TestCase
{
    public function testEncodePayload(): void
    {
        $payload = ['id' => '123', 'name' => 'test'];
        $encoded = Auth::encodePayload($payload);

        $this->assertIsString($encoded);
        $this->assertEquals($payload, json_decode(base64_decode($encoded), true));
    }

    public function testCreateSignature(): void
    {
        $sharedSecret = 'test_secret';
        $encodedPayload = 'dGVzdF9wYXlsb2Fk'; // base64 encoded "test_payload"

        $signature = Auth::createSignature($sharedSecret, $encodedPayload);

        $this->assertIsString($signature);
        $this->assertEquals(64, strlen($signature)); // SHA256 produces 64 char hex string
    }

    public function testVerifySignature(): void
    {
        $sharedSecret = 'test_secret';
        $encodedPayload = 'dGVzdF9wYXlsb2Fk';
        $signature = Auth::createSignature($sharedSecret, $encodedPayload);

        $this->assertTrue(Auth::verifySignature($sharedSecret, $encodedPayload, $signature));
        $this->assertFalse(Auth::verifySignature($sharedSecret, $encodedPayload, 'invalid_signature'));
    }

    public function testCreateAuthHeaders(): void
    {
        $accountId = 'test_account';
        $sharedSecret = 'test_secret';
        $payload = ['id' => '123'];

        $headers = Auth::createAuthHeaders($accountId, $sharedSecret, $payload);

        $this->assertArrayHasKey('X-ACCT-ID', $headers);
        $this->assertArrayHasKey('X-PAYLOAD-SIG', $headers);
        $this->assertArrayHasKey('Content-Type', $headers);
        $this->assertEquals($accountId, $headers['X-ACCT-ID']);
        $this->assertEquals('application/json', $headers['Content-Type']);
    }

    public function testCreateAuthHeadersWithoutPayload(): void
    {
        $accountId = 'test_account';
        $sharedSecret = 'test_secret';

        $headers = Auth::createAuthHeaders($accountId, $sharedSecret);

        $this->assertArrayHasKey('X-ACCT-ID', $headers);
        $this->assertArrayHasKey('X-PAYLOAD-SIG', $headers);
    }

    public function testCreateGetAuthHeaders(): void
    {
        $accountId = 'test_account';
        $sharedSecret = 'test_secret';
        $sigPayload = ['id' => '123'];

        $result = Auth::createGetAuthHeaders($accountId, $sharedSecret, $sigPayload);

        $this->assertArrayHasKey('headers', $result);
        $this->assertArrayHasKey('sigPayload', $result);
        $this->assertArrayHasKey('X-ACCT-ID', $result['headers']);
        $this->assertArrayHasKey('X-PAYLOAD-SIG', $result['headers']);
        $this->assertIsString($result['sigPayload']);
    }
}

File: packages/java-sdk/src/test/java/io/wusul/sdk/auth/AuthTest.java
package io.wusul.sdk.auth;

import org.junit.jupiter.api.Test;
import java.util.HashMap;
import java.util.Map;
import static org.junit.jupiter.api.Assertions.*;

/**
 * Tests for the Auth utilities.
 */
public class AuthTest {

    @Test
    public void testEncodePayload() {
        Map<String, String> payload = new HashMap<>();
        payload.put("id", "test");

        String encoded = Auth.encodePayload(payload);
        assertNotNull(encoded);
        assertFalse(encoded.isEmpty());
    }

    @Test
    public void testCreateSignature() {
        String sharedSecret = "test-secret";
        String encodedPayload = "dGVzdA==";

        String signature = Auth.createSignature(sharedSecret, encodedPayload);
        assertNotNull(signature);
        assertEquals(64, signature.length()); // SHA-256 produces 64 hex characters
    }

    @Test
    public void testVerifySignature() {
        String sharedSecret = "test-secret";
        String encodedPayload = "dGVzdA==";

        String signature = Auth.createSignature(sharedSecret, encodedPayload);
        assertTrue(Auth.verifySignature(sharedSecret, encodedPayload, signature));
        assertFalse(Auth.verifySignature(sharedSecret, encodedPayload, "invalid-signature"));
    }

    @Test
    public void testCreateAuthHeaders() {
        Map<String, String> headers = Auth.createAuthHeaders("account-id", "shared-secret", null);

        assertNotNull(headers);
        assertTrue(headers.containsKey("X-ACCT-ID"));
        assertTrue(headers.containsKey("X-PAYLOAD-SIG"));
        assertTrue(headers.containsKey("Content-Type"));
        assertEquals("account-id", headers.get("X-ACCT-ID"));
        assertEquals("application/json", headers.get("Content-Type"));
    }

    @Test
    public void testCreateGetAuthHeaders() {
        Auth.AuthHeadersResult result = Auth.createGetAuthHeaders("account-id", "shared-secret", null);

        assertNotNull(result);
        assertNotNull(result.getHeaders());
        assertNotNull(result.getSigPayload());
        assertTrue(result.getHeaders().containsKey("X-ACCT-ID"));
        assertTrue(result.getHeaders().containsKey("X-PAYLOAD-SIG"));
    }
}
